drracket/drracket/private/syncheck/gui.rkt

```diff
@@ -400,1671 +399,1601 @@ If the namespace does not, they are colored the unbound color.
-            ;; arrow-entry is one of
-            ;;   - (cons (U #f sym) (menu -> void))
-            ;;   - def-link
-            ;;   - tail-link
-            ;;   - arrow
-            ;;   - string
-            ;;   - colored-region
-            (define/private (get-arrow-record text)
-              (unless (object? text)
-                (error 'get-arrow-record "expected a text as the second argument, got ~e" text))
-              (hash-ref! arrow-records text (lambda () (make-interval-map))))
-
-            (define arrow-records #f)
-
-            (define/private (fetch-arrow-records txt pos)
-              (and arrow-records
-                   (let ([im (hash-ref arrow-records txt #f)]) 
-                     (if im
-                         (interval-map-ref im pos '())
-                         '()))))
-            
-            (define/public (dump-arrow-records)
-              (cond
-                [arrow-records
-                 (for ([(k v) (in-hash arrow-records)])
-                   (printf "\n\n~s:\n" k)
-                   (let loop ([it (interval-map-iterate-first v)])
-                     (when it
-                       (printf "~s =>\n" (interval-map-iterate-key v it))
-                       (for ([v (in-list (interval-map-iterate-value v it))])
-                         (printf "  ~s\n" v))
-                       (printf "\n")
-                       (loop (interval-map-iterate-next v it)))))]
-                [else
-                 (printf "arrow-records empty\n")]))
-            
-            ;; cleanup-texts : (or/c #f (listof text))
-            (define cleanup-texts #f)
-            
-            ;; definition-targets : hash-table[(list symbol[id-name] (listof symbol[submodname])) 
-            ;;                                 -o> (list text number number)]
-            (define definition-targets (make-hash))
-
-            ;; bindings-table : hash-table[(list text number number)
-            ;;                             -o> (setof (list text number number))]
-            ;; this is a private field
-            (define bindings-table (make-hash))
-
-            ;; unused-require-table : hash-table[(list text number number) -o> #t]
-            ;; this table records if a given require appears to be unused
-            (define unused-require-table (make-hash))
-            
-            ;; add-to-bindings-table : text number number text number number -> boolean
-            ;; results indicates if the binding was added to the table. It is added, unless
-            ;;  1) it is already there, or
-            ;;  2) it is a link to itself
-            (define/private (add-to-bindings-table start-text start-left start-right
-                                                   end-text end-left end-right)
+    (define (make-syncheck-text% super%)
+      (define cursor-arrow (make-object cursor% 'arrow))
+      (class*
+       (docs-text-defs-mixin super%)
+       (syncheck-text<%>)
+       (inherit set-cursor
```

Comment:
in class*, I'd like the keep the super class and the super interfaces on the first line and then have a newline. Is that possible @sorawee ?

Reply:
Totally! See https://github.com/sorawee/fmt/issues/73